### ISSUE TITLE
Add method to check if soft buttons are present

### ISSUE THREAD

**Issue by wmonecke** [Sunhat/react-native-extra-dimensions-android#41]
Hey!

I think for this library it would be really handy to have a method that return true or false wether the soft buttons are active or not since it is not possible to know with react native alone.

This is the code I have found to create such method:

DetectHardwareModule.java 

```
package com.companyname.appname.detecthardware;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.Callback;
import com.facebook.react.bridge.ReactMethod;

import java.util.List;

public class DetectHardwareModule extends ReactContextBaseJavaModule {

    public DetectHardwareModule(ReactApplicationContext reactContext) {
        super(reactContext);
    }

    @Override
    public String getName() {
        return "DetectHardware";
    }

    @ReactMethod
    public void hasHardwareButtons(final Callback callback) {
        callback.invoke(ViewConfiguration.get(getReactApplicationContext()).hasPermanentMenuKey());
    }
}
```

DetectHardwarePackage.java

```
package com.companyname.appname.detecthardware;

import com.facebook.react.ReactPackage;
import com.facebook.react.bridge.JavaScriptModule;
import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.uimanager.ViewManager;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class DetectHardwarePackage implements ReactPackage {
    @Override
    public List<NativeModule> createNativeModules(ReactApplicationContext reactContext) {
        List<NativeModule> modules = new ArrayList<>();
        modules.add(new DetectHardwareModule(reactContext));
        return modules;
    }

    @Override
    public List<Class<? extends JavaScriptModule>> createJSModules() {
        return Collections.emptyList();
    }

    @Override
    public List<ViewManager> createViewManagers(ReactApplicationContext reactContext) {
        return Collections.emptyList();
    }
}
```

**Comment by sebqq** [comment]
+1

**Comment by smilingkylan** [comment]
+1 it seems that when I use certain Android devices that it tells me the height of the soft menu bar as being greater than zero **even if the soft menu bar is not even being displayed**

An example would be the Google Nexus 5 (6.0 API 23). This forces our team to use special code for these devices which is a huge pain. Would love a fix for this!

PS Thank you for maintaining this repo, it has been incredibly useful for our team.

**Comment by Sunhat** [comment]
Hi there, could you submit a PR for this?

**Comment by Sunhat** [comment]
> +1 it seems that when I use certain Android devices that it tells me the height of the soft menu bar as being greater than zero **even if the soft menu bar is not even being displayed**
> 
> An example would be the Google Nexus 5 (6.0 API 23). This forces our team to use special code for these devices which is a huge pain. Would love a fix for this!
> 
> PS Thank you for maintaining this repo, it has been incredibly useful for our team.

Welcome to the hell that is React Native

**Comment by Sunhat** [comment]
Incoming...

**Comment by Sunhat** [comment]
Fixed in 1.1
